### ISSUE TITLE
Make it convenient to get return values from OSC server.

### ISSUE THREAD

**Issue by crocket** [attwad/python-osc#104]
https://github.com/njh/jackminimix returns values to OSC server calls.

Thus, I have to write

```
>>> import pythonosc.udp_client as uc
>>> client=uc.SimpleUDPClient("localhost", 14629)
>>> client.send(uc.OscMessageBuilder("/mixer/get_channel_count").build())
>>> res=uc.OscMessage(client._sock.recv(4096))
>>> res.address
'/mixer/channel_count'
>>> res.params
[1]
```

Getting return values should be more straightforward than using a hidden instance variable such as `_sock`.

**Comment by TheOnlyJoey** [comment]
I agree, I am working on building a pure client for receiving OSC information (like play/pause information in a DAW), but of course this is done by connecting to the client, hence needing to just passively receive. Would be nice being able to do this in such a fashion as the server implementation

**Comment by TheOnlyJoey** [comment]
@crocket It seems that the info from the udp socket is not even convertible to a OscMessage

> Traceback (most recent call last):
>   File "/usr/local/lib/python3.6/dist-packages/pythonosc/osc_message.py", line 32, in _parse_datagram
>     type_tag, index = osc_types.get_string(self._dgram, index)
>   File "/usr/local/lib/python3.6/dist-packages/pythonosc/parsing/osc_types.py", line 70, in get_string
>     'OSC string cannot begin with a null byte: %s' % dgram[start_index:])
> pythonosc.parsing.osc_types.ParseError: OSC string cannot begin with a null byte: b'\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x10/stop\x00\x00\x00,f\x00\x00?\x80\x00\x00\x00\x00\x00\x10/pause\x00\x00,f\x00\x00?\x80\x00\x00\x00\x00\x00\x10/play\x00\x00\x00,f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10/time\x00\x00\x00,f\x00\x00B\x1b\xcdV\x00\x00\x00\x1c/time/str\x00\x00\x00,s\x00\x000:38.950\x00\x00\x00\x00\x00\x00\x00\x18/beat/str\x00\x00\x00,s\x00\x0020.2.90\x00\x00\x00\x00\x14/samples\x00\x00\x00\x00,f\x00\x00I\xd1\xae\xb0\x00\x00\x00\x1c/samples/str\x00\x00\x00\x00,s\x00\x001717718\x00\x00\x00\x00\x1c/frames/str\x00,s\x00\x000:00:38:28\x00\x00

`

**Comment by DrLuke** [comment]
The provided servers work with a dispatcher. You have to map callback functions to addresses you want to receive at. See the [example](https://github.com/attwad/python-osc/blob/master/examples/dispatcher.py).

Tip: You can also map the method of an object!

**Comment by bobh66** [comment]
@crocket see the changes in #173 released in 1.9.0 that support bi-directional communication on the same port.

**Comment by crocket** [comment]
Closing it.
